src/btree/bt_compact.c

```diff
@@ -391,6 +391,22 @@ __wt_compact(WT_SESSION_IMPL *session)
             WT_WITH_PAGE_INDEX(session, ret = __compact_walk_internal(session, ref));
         }
 
+        if (ret == EBUSY) {
```

Comment:
> So this condition should be in the same if block as that call, a few lines up.

Yes, that's the intention. Good point!

> then move on to the next page without ever compacting the page that generated the EBUSY.

That's also a good point! I think this is a less likely case since compact will do another pass if it:

- moved at least 1 page.
- there is still room to move.

Then during the next pass, it would hopefully walk the internal page that it abandoned.  But this is not guaranteed so I will refactor with your suggestion. Thanks!